### ISSUE TITLE
inconsistent treating of UNSET by conditional natives

### ISSUE THREAD

**Issue by 9214** [red/red#4482]
**Describe the bug**
Some natives treat `unset` as truthy:

```red
>> if () [1]
== 1
>> unless () [1]
== none
>> any [()]
>> all [()]
>> not ()
== false
```

While others don't even recognize it as a valid conditional value:

```red
>> either () [1][2]
*** Script Error: block did not return a value
*** Where: either
*** Stack:  

>> until []
*** Script Error: block did not return a value
*** Where: until
*** Stack: 
 
>> while [][]
*** Script Error: block did not return a value
*** Where: while
*** Stack: 

>> case [() 1]
*** Script Error: block did not return a value
*** Where: case
*** Stack:  
```

The culprit is `logic/true?` that has this line inside it:
https://github.com/red/red/blob/2417a84e2cefce5f365d6ae675be994e04328a3f/runtime/datatypes/logic.reds#L72

**To reproduce**
See above.

**Expected behavior**
I expect the behavior to be consistent across all the natives, either one way or another.

**Platform version**

`master`, 2417a84e

**Comment by greggirwin** [comment]
I wouldn't jump to change this too quickly. You have new doc strings in #4483, presuming a change to allow unset for all of them, but I'm not sure that's best. In R2, the `if/unless/not` error thus:
```
>> if () []
** Script Error: if is missing its condition argument
** Near: if () []
>> unless () []
** Script Error: unless is missing its condition argument
** Near: unless () []
>> not ()
** Script Error: not is missing its value argument
** Near: not ()
```
So `any/all` are different, while the others all error, which I think is OK. Having different errors is OK too, as they are specific to block/non-block scenarios.

Unset is a strange beast, and we need to treat it as such. `Any/all` are different use cases for it, where all legitimate values have to be considered.

**Comment by 9214** [comment]
@greggirwin no, not at all. #4483 merely casts in code what we discussed so far in #4469; it can be refined as we proceed further, esp. in the light of this ticket, which I opened while working on #4484, which in turn proposes an alternative semantics discussed in https://github.com/red/red/issues/4469#issuecomment-635464683. Both PRs are meant for discussion and evaluation, nothing is set in stone.

Also, this is surely a place where we can learn the hard lessons of ornery error-throwing `money!`. I agree that `any` and `all` should permit `unset`, but that makes phrasing of their docstrings... tricky.

**Comment by hiiamboris** [comment]
https://github.com/red/red/blob/c2f0d8999dbd79ef458cfbe6e5d7c42f46d8b1fd/runtime/datatypes/logic.reds#L72
This is a curious distinction indeed...

**Comment by hiiamboris** [comment]
Note:
R2 allows unset for `all`/`any`, but not for `if`/`either`/`case`
R3 allows it for `if`, `either`, `all`/`any`, but not for `case`
Red allows it for `if`, `all`/`any` but not for `either`/`case`
